### DISCUSSION TITLE
Sensors access permission

### DISCUSSION THREAD

**Discussion by Ilithy** [florisboard/florisboard#1516] [Q&A] [answered]
Hello,
I realized, while checking the permissions assigned to Florisboard, that it had access to the sensors of my device (under Android 12).

_Sorry for the french screenshot_
![_20220120_172807](https://user-images.githubusercontent.com/36798218/150380296-e0ecee9d-48b9-40e2-afd4-e0d8381a40e1.JPG)

I have looked at the manifest file, and there is no mention of the request for permission to access the sensors.

In this case, is it possible for florisboard to refuse this access? (To clearly indicate to the OS not to allocate this access)?

Thank you for this excellent application

**Comment by patrickgold** [comment, marked as answer]
Thanks for bringing this up! I have not received any reports regarding this issue before nor is this issue reproducible on any device, GPlay and F-Droid also do not detect that this permission is required for FlorisBoard beta (0.3.14-beta08), which is correct.

What is the exact list of permissions FlorisBoard needs according to your device settings app? The only theory I have is that requiring vibration magically adds the sensor permission, but this would be a complete new one for me. Also, besides stating your Android version, could you please state your device model and if you use the stock ROM of the manufacturer or a custom ROM? Thanks in advance!

> In this case, is it possible for florisboard to refuse this access? (To clearly indicate to the OS not to allocate this access)?

When I have a bit more info I can take a look if it is possible to add a specific deny permission entry to the manifest for sensors.

**Comment by Ilithy** [comment]
Of course, thank you for responding so quickly.
Device: Pixel 6 pro MP1.0
Custom OS: Grapheneos - (based on Android 12) Build SQ1D.220105.007.2022011423
(Secure boot: yes)
App version: Florisboard beta version 0.3.14.beta08

Before seeing this famous permission, Florisboard was set to use the vibration,
I have the vibration active, even after removing the permission of access to the sensors (they are dissociate, see my screenshot below), Florisboard requires therefore only to the vibration permission access.
 
_Here is the complete list of possible authorizations for Florisboard_
_The second section is: "Other application features" including the "Use vibration" permission_
![IMG_20220120_191428](https://user-images.githubusercontent.com/36798218/150398983-3bccdec4-c124-41a9-8b99-d33785842b87.jpg)


I prefer to specify, I do not accuse Florisboard of anything, This is an wonderful and safe application;
I just ask the question, it is not a problem in itself, but the less applications have authorizations, the less risks there are. (At least it seems to me)
Thanks again

**Reply by Ilithy** [reply]
It seems that Grapheneos has added a specific authorization to the sensor, so this authorization is not present in aosp

"the Sensors app permission added by GrapheneOS can be toggled off for the browser app as a whole instead."

"Some of these features are always enabled under the hood, while others like the Network and Sensors toggles are controlled by users via opt-in or opt-out toggles. Apps may not handle having access taken away like this, although it generally doesn't cause any issues as it's all designed to be friendly to apps and fully compatible rather than killing the application when it violates the rules."

Source: https://grapheneos.org/usage

**Reply by patrickgold** [reply]
Thanks for giving this info, it really helped finding the cause:

GrapheneOS adds a sensors permission to all apps so the user can disable sensors for an app. This permission is not existent in AOSP or other Android ROMs, so this means normally I would not even have to ask permission to read out sensors in the first place. This also explains why I don't see the permission on any of my test devices.

Sadly there does not seem to be a way for me to declare that I want to auto-deny a permission, so I think you just should disable the permission (FlorisBoard does not need sensor data) and live with it.

Hope this satisfies your request!

**Reply by Ilithy** [reply]
Thanks, I suspected that you won't be able to do anything (due to the specific functioning of Grapheneos), and indeed, I just have to remove the authorization in the settings (the behavior of Florisboard is not altered), thanks a lot.
